assemblies/assembly_openshift-operator/main.adoc

```diff
@@ -0,0 +1,46 @@
+[id='assembly_openshift-operator']
+
+include::_artifacts/document-attributes.adoc[]
+
+= Deploying a {PRODUCT} {PRODUCT_VERSION} environment on {OPENSHIFT} using Operators
+// Context attribute is assembly specific and enables module reuse between assemblies.
+:doctype: book
+:imagesdir: _images
+:context: openshift-operator
+include::_artifacts/author-group.adoc[]
+
+// Purpose statement for the assembly
+As a system engineer, you can deploy a {PRODUCT} environment on {OPENSHIFT} to provide an infrastructure to develop or execute processes and other business assets. You can use OpenShift Operators to deploy the environment defined in a structured YAML file and to maintain and modify this environment as necessary.
+
+.Prerequisites
+
+* At least four gigabytes of memory must be available in the OpenShift environment.
+* The OpenShift Operator Framework must be installed and started in the OpenShift environment.
+* The OpenShift project for the deployment must be created.
+* You must be logged in to the project using the OpenShift web console 
+* Dynamic persistent volume (PV) provisioning must be enabled. Alternatively, if dynamic PV provisioning is not enabled,  enough persistent volumes must be available. By default, the following sizes are required:
+** Each deployed replicated set of {KIE_SERVER} pods, by default, requires one 1Gi PV for the database. You can change the database PV size in the playbook parameters. You can deploy multiple immutable servers; each requires a separate database PV. This requirement does not apply if you use an external database server.
+** By default, {CENTRAL} requires one 1Gi PV. You can change the PV size for {CENTRAL} persistent storage in the playbook parameters.
+** {CENTRAL} Monitoring requires one 64Mi PV.
+** Smart Router requires one 64Mi PV.
+* If you intend to scale any of the {CENTRAL} or {CENTRAL} Monitoring pods, your OpenShift environment must support persistent volumes with ReadWriteMany mode.
++
+IMPORTANT: ReadWriteMany mode is not supported on OpenShift Online and OpenShift Dedicated.
+
+include::{enterprise-dir}/openshift/ba-openshift-overview-con.adoc[leveloffset=+1]
+include::{enterprise-dir}/openshift/dm-openshift-prepare-con.adoc[leveloffset=+1]
+include::{enterprise-dir}/openshift/registry-authentication-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/openshift/secrets-kie-create-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/openshift/secrets-central-create-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/openshift/secrets-smartrouter-create-proc.adoc[leveloffset=+2]
```

Comment:
do we need any of the above secrets includes with the operator? we don't require them, but its certainly an option for users.

Reply:
Secrets are needed as long as they are required for production